src/vm/eventtrace.cpp

```diff
@@ -1783,18 +1786,60 @@ void BulkTypeEventLogger::FireBulkTypeEvent()
     }
 
     Win32EventWrite(Microsoft_Windows_DotNETRuntimeHandle, &BulkType, iDesc, EventData);
+    
+#else
 
+    UINT iSize = 0;
+    
+    for (int iTypeData = 0; iTypeData < m_nBulkTypeValueCount; iTypeData++)
+    {
+        // Do fixed-size data as one bulk copy
+        memcpy(
+                m_BulkTypeEventBuffer + iSize,
+                &(m_rgBulkTypeValues[iTypeData].fixedSizedData),
+                sizeof(m_rgBulkTypeValues[iTypeData].fixedSizedData));
+        iSize += sizeof(m_rgBulkTypeValues[iTypeData].fixedSizedData);
+
+        // Do var-sized data individually per field
+
+        LPCWSTR wszName = m_rgBulkTypeValues[iTypeData].sName.GetUnicode();
+        if (wszName == NULL)
+        {
+            m_BulkTypeEventBuffer[iSize++] = 0;
+            m_BulkTypeEventBuffer[iSize++] = 0;
+        }
+        else
+        {
+            UINT nameSize = (m_rgBulkTypeValues[iTypeData].sName.GetCount() + 1) * sizeof(WCHAR);
+            memcpy(m_BulkTypeEventBuffer + iSize, wszName, nameSize);
+            iSize += nameSize;
+        }
+
+        // Type parameter count
+        ULONG params = m_rgBulkTypeValues[iTypeData].rgTypeParameters.GetCount();
+        
+        ULONG *ptrInt = (ULONG*)(m_BulkTypeEventBuffer + iSize);
+        *ptrInt = params;
+        iSize += 4;
```

Comment:
So we actually need to write 4 bytes to the stream and increment iSize by exactly 4.  This is the format of the ETW buffer that we are copying here.  We specifically do NOT want to use sizeof(ULONG), and definitely use "iSize += 4".  In this case it doesn't matter if ULONG is 32 bits or 64, as long as the platform is little endian.  We'll write the 4 bytes we care about to the correct place...  We could change ULONG to be uint32_t (#include <stdint.h>) instead to make that more clear.